src/main/java/graphql/nadel/engine/OverallQueryTransformer.java

```diff
@@ -403,7 +403,13 @@ public TraversalControl enter(TraverserContext<Node> context) {
                 service,
                 serviceContext);
         Map<Class<?>, Object> rootVars = new LinkedHashMap<>();
-        GraphQLOutputType underlyingSchemaParent = (GraphQLOutputType) underlyingSchema.getType(parentType.getName());
+        TypeMappingDefinition mappingDefinition = getTypeMappingDefinitionFor(parentTypeOverall);
+        String underlyingParentName = parentTypeOverall.getName();
+        if (mappingDefinition != null) {
+            typeRenameMappings.put(mappingDefinition.getUnderlyingName(), mappingDefinition.getOverallName());
+            underlyingParentName = mappingDefinition.getUnderlyingName();
+        }
```

Comment:
Can we put this fairly eseoteric code in its own method (for maintaince reasons)

Reply:
done